### PR TITLE
Verify last_update_check response conditionally

### PR DESCRIPTION
**Pull request by GGP1**
|Related issue|
|---|
| Closes https://github.com/wazuh/wazuh/issues/21822 |

## Description

Modifies the function used to validate last update check responses to consider failures because of the CTI service being unavailable for unreleased versions.

### Tests

> The check failed because https://github.com/wazuh/wazuh/issues/21743 is not fixed in 4.8.1 yet

<details><summary>test_manager_endpoints.tavern.yaml</summary>

> I had to make some modifications to make all stages run successfully but they are not related to the issue so I didn't include them.
>
> It's expected that the tests fail in the check.

```console
(venv10) gasti@pop-os:~/work/wazuh/api/test/integration$ pytest -vv test_manager_endpoints.tavern.yaml --nobuild
====================================================== test session starts =======================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.10.12', 'Platform': 'Linux-6.6.10-76060610-generic-x86_64-with-glibc2.35', 'Packages': {'pytest': '7.3.1', 'pluggy': '1.0.0'}, 'Plugins': {'trio': '0.8.0', 'asyncio': '0.18.1', 'tavern': '1.23.5', 'html': '2.1.1', 'aiohttp': '1.0.4', 'metadata': '3.0.0'}}
rootdir: /home/gasti/work/wazuh/api/test/integration
configfile: pytest.ini
plugins: trio-0.8.0, asyncio-0.18.1, tavern-1.23.5, html-2.1.1, aiohttp-1.0.4, metadata-3.0.0
asyncio: mode=auto
collected 35 items                                                                                                               

test_manager_endpoints.tavern.yaml::GET /manager/status PASSED                                                             [  2%]
test_manager_endpoints.tavern.yaml::GET /manager/info PASSED                                                               [  5%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[alerts] PASSED                                    [  8%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[auth] PASSED                                      [ 11%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[cis-cat] PASSED                                   [ 14%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[cluster] PASSED                                   [ 17%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[command] PASSED                                   [ 20%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[global] PASSED                                    [ 22%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[localfile] PASSED                                 [ 25%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[osquery] PASSED                                   [ 28%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[remote] PASSED                                    [ 31%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[rootcheck] PASSED                                 [ 34%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[ruleset] PASSED                                   [ 37%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[sca] PASSED                                       [ 40%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[syscheck] PASSED                                  [ 42%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[syscollector] PASSED                              [ 45%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[vulnerability-detection] PASSED                   [ 48%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[indexer] PASSED                                   [ 51%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration PASSED                                                      [ 54%]
test_manager_endpoints.tavern.yaml::GET /manager/daemons/stats PASSED                                                      [ 57%]
test_manager_endpoints.tavern.yaml::GET /manager/stats PASSED                                                              [ 60%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/hourly PASSED                                                       [ 62%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/weekly PASSED                                                       [ 65%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/analysisd PASSED                                                    [ 68%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/remoted PASSED                                                      [ 71%]
test_manager_endpoints.tavern.yaml::GET /manager/logs PASSED                                                               [ 74%]
test_manager_endpoints.tavern.yaml::GET /manager/logs/summary PASSED                                                       [ 77%]
test_manager_endpoints.tavern.yaml::GET /manager/api/config PASSED                                                         [ 80%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration/validation (OK) PASSED                                      [ 82%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration/{component}/{configuration} PASSED                          [ 85%]
test_manager_endpoints.tavern.yaml::PUT /manager/configuration PASSED                                                      [ 88%]
test_manager_endpoints.tavern.yaml::GET /manager/version/check PASSED                                                      [ 91%]
test_manager_endpoints.tavern.yaml::GET /manager/version/check with update_check disabled PASSED                           [ 94%]
test_manager_endpoints.tavern.yaml::GET /manager/version/check with update check service error PASSED                      [ 97%]
test_manager_endpoints.tavern.yaml::PUT /manager/restart PASSED                                                            [100%]

========================================= 35 passed, 36 warnings in 216.31s (0:03:36) ===========================================
```

</details>

### CODE REVIEW

File: api/test/integration/tavern_utils.py [hunk 1]
```diff
@@ -510,20 +510,36 @@ def healthcheck_agent_restart(response, agents_list):
     check_agent_active_status(agents_list)
 
 
-def validate_update_check_response(response):
-    """Check if the last_available_* dicts have the correct keys and values.
+def validate_update_check_response(response, current_version, update_check):
+    """Check that the update check response contains the fields expected and
```
Comment (EduLeon12) [inline api/test/integration/tavern_utils.py:514]:
```suggestion
    """Check that the update check response contains the expected fields, and verify if the 'last_available_*' dictionaries have the correct keys and values.
```
A small suggestion to improve the readability of the method docstring. 
Let me know what you think @GGP1.

**Review by EduLeon12 — CHANGES_REQUESTED** [review]
Requested small changes for improved readability.

**Review by EduLeon12 — APPROVED** [review]

**Review by Selutario — CHANGES_REQUESTED** [review]
Looks good! It seems that some of the `test_manager_endpoints` are failing though, could you check the reason?
- https://github.com/wazuh/wazuh/pull/21928/checks?check_run_id=21658376272

**Comment by GGP1** [comment]
> Looks good! It seems that some of the `test_manager_endpoints` are failing though, could you check the reason?
> 
> * https://github.com/wazuh/wazuh/pull/21928/checks?check_run_id=21658376272

The error shows `TLS/SSL connection has been closed`, that didn't happen when I executed them locally nor on the other runs though.

I've found a [question](https://stackoverflow.com/questions/75273639/caused-by-sslerrorsslzeroreturnerror6-tls-ssl-connection-has-been-closed-eo) in stackeroverflow and an [issue](https://github.com/googleapis/google-auth-library-python/issues/1356) describring this problem, but neither of them seem to be resolved. I think we should investigate it in a separate issue as it's not related to the changes introduced.

**Review by Selutario — APPROVED** [review]
Agree! Please, open it when possible. LGTM.

**Pull request merged**